### PR TITLE
Archiving cleanup

### LINKED ISSUE #2612: Disable archiving on half cycles and clean up yamls
### What new functionality do you need?

Archiving should be disabled on the first half cycle since the only data generated is the gdas forecast, which is not particularly useful.  Also, the yamls could be cleaned up and documentation/comments could be added as well.

### What are the requirements for the new functionality?

No data is stored in the ARCDIR or ATARDIR on the first half cycle and the gdasarch and enkfgdasearc jobs are not created in the rocoto XML for the first half cycle.

### Acceptance Criteria

Data archived after the first half cycle is identical to current develop, with the exception of dates, etc, in the log file.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# Description
1) Adds a lot of comments to the jinja templates for archiving
2) Rearranges the gdas and enkf templates to a more logical order
3) Fixes a couple of bugs in the enkf archiving of increments and analyses
4) Disables archiving for the half cycle
5) Removes the `FITSARC` key from `config.base` and `arcdir.yaml.j2`, instead relying on `DO_FIT2OBS`
6) Updates wxflow to add the option to not allow undefined variables when parsing jinja templates and invokes this feature when running archives

Resolves #2612

# Type of change
- Bug fix (fixes something broken)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Cycled test on Hera

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
A test run with hash 7d2c539f45194cd4e5b21bfd4b83a9480189cd0f (May 21) had `arch` fail on the first half-cycle with:
```
  File "/scratch1/NCEPDEV/da/Andrew.Eichmann/fv3gfs/develop/global-workflow/ush/python/pygfs/task/archive.py", line 199, in _create_fileset
    raise FileNotFoundError(f"FATAL ERROR: Required file, directory, or glob {item} not found!")
FileNotFoundError: FATAL ERROR: Required file, directory, or glob gdas.20210630/00/model_data/ocean/history/gdas.ocean.t00z.inst.f000.nc not found!
```
In the two subsequent cycles `arch` fails with the following:
```
  File "/scratch1/NCEPDEV/da/Andrew.Eichmann/fv3gfs/develop/global-workflow/ush/python/pygfs/task/archive.py", line 199, in _create_fileset
    raise FileNotFoundError(f"FATAL ERROR: Required file, directory, or glob {item} not found!")
FileNotFoundError: FATAL ERROR: Required file, directory, or glob gdas.20210630/06/analysis/ocean/gdas.t06z.ocn.adt_rads_all.stats.csv not found!
```
@guillaumevernieres tells me that the `csv` file in question is produced only on 00 cycles.

**Comment by guillaumevernieres** [comment]
@DavidHuber-NOAA , there's a bug in the archiving of the `csv` files.
applying this patch should fix the issue:
```diff
diff --git a/parm/archive/gdasocean_analysis.yaml.j2 b/parm/archive/gdasocean_analysis.yaml.j2
index 0c43cd40..042f434e 100644
--- a/parm/archive/gdasocean_analysis.yaml.j2
+++ b/parm/archive/gdasocean_analysis.yaml.j2
@@ -9,7 +9,7 @@ gdasocean_analysis:
         {% for domain in ["ocn", "ice"] %}
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}{{domain}}.bkgerr_stddev.nc'
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}{{domain}}.incr.nc'
-        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}{{domain}}ana.nc'
+<        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}{{domain}}ana.nc'
         {% if NMEM_ENS > 2 %}
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}{{domain}}.recentering_error.nc'
         {% endif %}
@@ -20,8 +20,6 @@ gdasocean_analysis:
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}ocn.ssh_total_stddev.nc'
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}ocn.steric_explained_variance.nc'
         {% endif %}
-        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}ocn.adt_rads_all.stats.csv'
-        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}ocn.icec_amsr2_north.stats.csv'
-        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/{{ head }}ocn.icec_amsr2_south.stats.csv'
+        - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/*.stats.csv'
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/diags/*.nc4'
         - '{{ COM_OCEAN_ANALYSIS | relpath(ROTDIR) }}/yaml/*.yaml'
```

**Comment by DavidHuber-NOAA** [comment]
Thanks for the heads up @AndrewEichmann-NOAA @guillaumevernieres.  This PR will disable archiving for the first half cycle, so that should resolve the missing `gdas.ocean.t00z.inst.f000.nc` error.

For the csv files, I would like to test out the suggested change.  Could you share your experiment setup so I can run a test case?

**Comment by guillaumevernieres** [comment]
> Thanks for the heads up @AndrewEichmann-NOAA @guillaumevernieres. This PR will disable archiving for the first half cycle, so that should resolve the missing `gdas.ocean.t00z.inst.f000.nc` error.
> 
> For the csv files, I would like to test out the suggested change. Could you share your experiment setup so I can run a test case?

No problem and thanks for doing this work. This was discovered while running the c384/0.25, but that could be tested at lower res with the existing test, you'll just need to extend the date since it only does 1.5 cycle:

https://github.com/NOAA-EMC/global-workflow/blob/7d2c539f45194cd4e5b21bfd4b83a9480189cd0f/ci/cases/pr/C48mx500_3DVarAOWCDA.yaml#L14

**Comment by guillaumevernieres** [comment]
Oops ... 1 more thing @DavidHuber-NOAA , turns out we don't have enough marine obs in glopara to do enough cycles to trigger the archiving.
You'll have to update `DMPDIR`:
https://github.com/NOAA-EMC/global-workflow/blob/7d2c539f45194cd4e5b21bfd4b83a9480189cd0f/parm/config/gfs/yaml/defaults.yaml#L55

to

```yaml
DMPDIR: /scratch1/NCEPDEV/da/common/
```

**Comment by DavidHuber-NOAA** [comment]
OK, thanks for the heads up @guillaumevernieres.

**Comment by DavidHuber-NOAA** [comment]
@guillaumevernieres Alright, I have a fix in for the csv files.

File: ush/python/pygfs/task/archive.py [hunk 1]
```diff
@@ -59,10 +59,22 @@ def configure(self, arch_dict: Dict[str, Any]) -> (Dict[str, Any], List[Dict[str
             List of tarballs and instructions for creating them via tar or htar
         """
 
+        if not os.path.isdir(arch_dict.ROTDIR):
+            raise FileNotFoundError(f"FATAL ERROR: The ROTDIR ({arch_dict.ROTDIR}) does not exist!")
+
+        if arch_dict.RUN == "gefs":
+            raise NotImplementedError("FATAL ERROR: Archiving is not yet set up for GEFS runs")
+
+        if arch_dict.RUN == "gdas" or arch_dict.RUN == "gfs":
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/task/archive.py]:
```suggestion
        if arch_dict.RUN in ["gdas", "gfs"]:
```

File: ush/python/pygfs/task/archive.py [hunk 2]
```diff
@@ -72,7 +84,16 @@ def configure(self, arch_dict: Dict[str, Any]) -> (Dict[str, Any], List[Dict[str
...
 
         # Parse the input jinja yaml template
-        arcdir_set = parse_j2yaml(arcdir_filename, arch_dict)
+        arcdir_set = Archive._construct_arcdir_set(arcdir_j2yaml,
+                                                   arch_dict)
+
+        if not os.path.isdir(arch_dict.ROTDIR):
+            raise FileNotFoundError(f"FATAL ERROR: The ROTDIR ({arch_dict.ROTDIR}) does not exist!")
+
+        if arch_dict.RUN == "gdas" or arch_dict.RUN == "gfs":
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/task/archive.py]:
```suggestion
        if arch_dict.RUN in ["gdas", "gfs"]:
```

File: ush/python/pygfs/task/archive.py [hunk 3]
```diff
@@ -312,6 +323,22 @@ def _gen_relative_paths(self, root_path: str) -> Dict:
 
         return rel_path_dict
 
+    @staticmethod
+    @logit(logger)
+    def _construct_arcdir_set(j2yaml, arch_dict) -> Dict:
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/task/archive.py]: This should have a doc-block

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: ush/python/pygfs/task/archive.py (hunk 1 above)
Comment (DavidHuber-NOAA) [reply ush/python/pygfs/task/archive.py]: Updated, thanks.

File: ush/python/pygfs/task/archive.py (hunk 2 above)
Comment (DavidHuber-NOAA) [reply ush/python/pygfs/task/archive.py]: Updated.

File: ush/python/pygfs/task/archive.py (hunk 3 above)
Comment (DavidHuber-NOAA) [reply ush/python/pygfs/task/archive.py]: Yes it should.  I've added a block and inline comments.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

I didn't check the jinja templates closely. I'm trusting they are producing the correct tarballs.

**Comment by DavidHuber-NOAA** [comment]
@TerrenceMcGuinness-NOAA It looks like the builds failed on Hera.  Could you reboot CI on that platform?

**Comment by DavidHuber-NOAA** [comment]
@TerrenceMcGuinness-NOAA Disregard, I see now that CI was restarted on Hera already.

**Pull request merged**